[One Workflow] Add concurrency group schema and ES mappings

h88 opened the pull request:
## Overview

This is a foundational PR that establishes the data model for concurrency groups and collision strategies without implementing any runtime logic. It sets up the schema definitions and Elasticsearch mappings required for subsequent sub-tasks.

## Changes

### Schema Changes (`spec/schema.ts`)
- Added `CollisionStrategySchema` enum with values: `'queue'`, `'drop'`, `'cancel-in-progress'`
- Added `CollisionStrategy` TypeScript type export
- Extended `WorkflowSettingsSchema` with three new optional fields:
  - `concurrency.key`: Optional String field for dynamic concurrency group key (e.g., `'{{ event.host.name }}'`)
  - `concurrency.strategy`: Optional Enum field
  - `concurrency.max`: Optional Integer field (min: 1)

### Elasticsearch Mappings (`common/mappings.ts`)
- Added `concurrencyGroupKey` as a top-level `keyword` field in `WORKFLOWS_EXECUTIONS_INDEX_MAPPINGS`
- This field will be used to efficiently query executions by concurrency group

### Testing (`spec/schema.test.ts`)
- Added comprehensive unit tests for `concurrency.key` validation
- Added tests for `concurrency.strategy` validation
- Added tests for `concurrency.max` validation
- Added tests for `CollisionStrategySchema` enum validation

## Expected Results (Acceptance Criteria)

- Schema validates workflows with `concurrency.key`, `concurrency.strategy`, and `concurrency.max` settings
- Schema rejects invalid `concurrency.strategy` values (must be 'queue', 'drop', or 'cancel-in-progress')
- Schema validates `concurrency.max` is a natural number
- Elasticsearch mapping includes `concurrencyGroupKey` as a searchable `keyword` field
- Monaco editor provides autocomplete for new settings fields (via JSON schema generation)
- TypeScript types are updated and compile without errors
- Unit tests validate schema constraints

## Notes

- This PR contains **no runtime logic** - only schema and infrastructure changes
- Monaco editor autocomplete will automatically work via JSON schema generation from Zod schemas
- "Default values (queue for concurrency.strategy, 1 for concurrency.max) will be applied at runtime in later sub-tasks that implement the concurrency logic"
- Subsequent PRs will implement the actual concurrency control logic


## Related Issues

Resolves https://github.com/elastic/security-team/issues/15054

## Screenshot

<img width="665" height="362" alt="image" src="https://github.com/user-attachments/assets/66a47910-ab59-4ee7-87d0-6755c9a0c0a2" />

h88 committed:
foundational schema definitions and Elasticsearch mappings for concurrency groups and collision strategies

h88 committed:
types & tests alignment

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask1-infra

Kiryous commented on src/platform/packages/shared/kbn-workflows/spec/schema.ts:
```diff
@@ -56,10 +56,16 @@ export function getOnFailureStepSchema(stepSchema: z.ZodType, loose: boolean = f
   'on-failure': WorkflowOnFailureSchema.optional(),
   timezone: z.string().optional(), // Should follow IANA TZ format
   timeout: DurationSchema.optional(), // e.g., '5s', '1m', '2h'
+  concurrency_key: z.string().optional(), // e.g., '{{ event.host.name }}'
```
NIT: we previously used kebab-case for yaml properties (e.g. `connector-id`, `on-failure`, `max-attempts`). idk if we want to continue to stick with kebab-case or migrate to snake-case? cc @shahargl @talboren @tinnytintin10

h88 committed:
consistency

Kiryous reviewed (approved):
LGTM

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask1-infra

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask1-infra

h88 committed:
refactor: nest concurrency settings under concurrency object

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/373339)
* Commit: c8b980e5a2fb991eb0f28f4b5aa6352fee05235d

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/373339#019b22ef-1f94-47e1-92a3-3dff33136d03)

### History
* :broken_heart: [Build #373171](https://buildkite.com/elastic/kibana-pull-request/builds/373171) failed 7b8be5e9cbab36688fd4658cce0185b6358ca4cf
* :yellow_heart: [Build #373050](https://buildkite.com/elastic/kibana-pull-request/builds/373050) was flaky 554ed663453e1d9b32e5444b9379cc42149163c4
* :broken_heart: [Build #372831](https://buildkite.com/elastic/kibana-pull-request/builds/372831) failed bf16d83182917da82864cb70c82f11e3d390cca7

cc @h88

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373339","number":373339,"commit":"c8b980e5a2fb991eb0f28f4b5aa6352fee05235d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373171","number":373171,"commit":"7b8be5e9cbab36688fd4658cce0185b6358ca4cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373050","number":373050,"commit":"554ed663453e1d9b32e5444b9379cc42149163c4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372831","number":372831,"commit":"bf16d83182917da82864cb70c82f11e3d390cca7"}],"number":373339}
buildkite-pr-comment-->

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask1-infra

skynetigor reviewed (approved):
LGTM

h88 committed:
Hide concurrency settings until feature is complete
Omit the `concurrency`  to prevent it from appearing in autocomplete (user facing) until all concurrency control sub-tasks are completed.

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask1-infra

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask1-infra
